### PR TITLE
Added new themes for lualine

### PR DESCRIPTION
**Pull request by VoxelPrismatic**
A couple of things here:

1. Moved the variants part of `/lua/rose-pine/palettes.lua` to another file
   - `palettes.lua` now imports `variants.lua`
   - I understand that it's annoying to jump around ten times, but it's not breaking!
2. Moved the lualine theme generators to a new file under `/lua/lualine/generate.lua`
   - This removes repeated code across the new themes
3. Created new theme files
   1. `rose-pine-moon` - Like `rose-pine`, but always uses the moon variant
   2. `rose-pine-main` - (etc...) main variant
   3. `rose-pine-dawn` - (etc...) dawn variant
   4. `rose-pine-alt-moon` - Like `rose-pine-alt`, but always uses the moon variant
   5. `rose-pine-alt-main` - (etc...) main variant
   6. `rose-pine-alt-dawn` - (etc...) dawn variant
 
Why? I just prefer a dark status line.
image

### CODE REVIEW

**Comment by mvllow** [comment]
Hey there! Appreciate the contribution—I do have some reservations, however. I think I'd prefer not having the variants pulled into a new file. Would it work/make sense to instead use `local p = require("rose-pine.palette").moon` to grab a specific variant's palette?

**Comment by VoxelPrismatic** [comment]
That's news to me!

Sorry, I'm not too familiar with Lua, so I'm happy to make those changes. I was under the impression that `rose-pine.pallete` only returned the theme set in the config

**Comment by VoxelPrismatic** [comment]
I just checked, and yes, my suspicions are correct. `require("rose-pine.palette")` only returns the palette as set by the config. `require("rose-pine.palette").moon` returns `nil`.

I suppose I can embed the other variants in the returned object, under `p._variants.moon` or something like that. Let me know if that's good with you.

**Comment by mvllow** [comment]
This all looks great, thank you! I do want to personally test everything before merging so it may be a few days :)

File: lua/rose-pine/palette.lua [hunk 1]
```diff
@@ -59,8 +60,14 @@ local variants = {
 	},
 }
 
-if variants[options.variant] ~= nil then
-	return variants[options.variant]
+
+local chosen
+if variants[options.dark_variant] ~= nil then
```
Comment (mvllow) [inline lua/rose-pine/palette.lua:65]:
```suggestion
if variants[options.variant] ~= nil then
```

**Review by mvllow — CHANGES_REQUESTED** [review]
Was getting an error on startup that `chosen` was undefined. Once changing the conditional it worked great :)

**Comment by mvllow** [comment]
Side note, I know you started this to be able to have a dark statusline in dawn (or perhaps light statusline in main/moon)—perhaps we could also add matching inverse themes, e.g. using `text` as the background.

Here's an example of an inverse theme for dawn:

Screenshot of Rosé Pine Dawn using a darker lualine theme within Neovim

**Comment by VoxelPrismatic** [comment]
At this point, I think we should find a way to do the following:
```lua
{
    theme = require("rose-pine.plugins.lualine").inverse.moon
}
```
Of course, passing "rose-pine", "rose-pine-alt", and "rose-pine-inv" will all work as expected. There are just too many names to keep track of and update. In this very example, we'd need to create three new files just for `inv`. And what happens when we add a new theme beyond Dawn, Moon and Main?

**Comment by VoxelPrismatic** [comment]
This latest commit creates a grey and inverse theme.
The grey theme follows your general idea, and looks like this:
image

The inverse theme actually chooses the opposite theme; main and moon will select normal:dawn, and dawn will select normal:moon.

I'm actually a big fan of the grey theme, and already switched to it.

**Comment by mvllow** [comment]
I like the grey, perhaps it would be better named "neutral"? As for the inverse, I think I prefer utilising the variant's palette only if possible.

Thank you for the iterations/changes :)

**Comment by VoxelPrismatic** [comment]
The inverse is specifically designed to pull the opposite color without much thought. Believe me, I tried to make a dark bar, but the colors get very difficult to see at smaller font sizes (I use 8px). Grey, or neutral, strikes a nice balance.

**Review by mvllow — APPROVED** [review]

**Pull request merged**